dehydrated-hook-ddns-tsig.py

```diff
@@ -152,10 +152,13 @@ def query_NS_record(domain_name):
     for i in range(0, len(name_list)):
         nameservers = []
         try:
-            for ns in [rdata.target.to_unicode()
-                       for rdata in dns.resolver.query('.'.join(name_list[i:]),
-                                                       'NS')]:
-                nameservers += [_.to_text() for _ in dns.resolver.query(ns)]
+            fqdn = '.'.join(name_list[i:])
+            for rdata in dns.resolver.query(fqdn, dns.rdatatype.NS):
+                ns = rdata.target.to_unicode()
+                nsL = [] 
+                nsL.extend([_.to_text() for _ in dns.resolver.query(ns)]) # default type: A
+                nsL.extend([_.to_text() for _ in dns.resolver.query(ns, rdtype=dns.rdatatype.AAAA)])
+                nameservers.append(nsL)
```

Comment:
this could be written more easily and probably with shorter execution time as:

    ns = rdata.target.to_unicode()
    nameservers.append(_.to_text() for _ in dns.resolver.query(ns)) # default type: A
    nameservers.append(_.to_text() for _ in dns.resolver.query(ns, rdtype=dns.rdatatype.AAAA))

Reply:
Yes, you could do that, but in case you run in a 30s timeout, you have no clue, where this comes from. I personally do not like lines that does multiple things and leave you in the dark if something goes wrong.